drivers/bluetooth/hci/hci_nxp.c

```diff
@@ -369,6 +472,13 @@ int bt_nxp_setup(const struct device *dev, const struct bt_hci_setup_params *par
 			break;
 		}
 #endif /* CONFIG_HCI_NXP_ENABLE_AUTO_SLEEP */
+#if CONFIG_BT_HCI_SET_PUBLIC_ADDR
+		ret = bt_nxp_set_mac_address();
```

Comment:
this would have to be #define bt_nxp_set_mac_address(...) above for the case when the config is not set if taking the other comment

Reply:
Done, Thanks